features/steps/steps.py

```diff
@@ -48,6 +50,55 @@ def fmt(x):
             return "...".join((v[:25], v[-7:]))
         return v
 
+def is_closed(seq_of_pt):
+    # @todo tolerance
+    return seq_of_pt[0] == seq_of_pt[-1]
+
+def get_points(inst, return_type = 'coord'):
+    if inst.is_a().startswith('IfcCartesianPointList'):
+        return inst.CoordList
+    elif inst.is_a('IfcPolyline'):
+        if return_type == 'coord':
+            return [p.Coordinates for p in inst.Points]
+        elif return_type == 'points':
+            return inst.Points
+    elif inst.is_a('IfcPolyLoop'):
+        if return_type == 'coord':
+            return [p.Coordinates for p in inst.Polygon]
+        elif return_type == 'points':
+            return inst.Polygon
+    else:
+        raise NotImplementedError(f'get_points() not implemented on {inst.is_a}')
+
+def get_entity_contexts(ifc_context, ent):
+    if hasattr(ent, 'ContextOfItems'):
+        return ent.ContextOfItems
+    set_instances = []
+    if len(ifc_context.model.get_inverse(ent)) == 0:
+        return None # inverse unattainable, context not found
+    for rev_item in ifc_context.model.get_inverse(ent):
+        if hasattr(rev_item, 'ContextOfItems'):
+            set_instances.append(rev_item.ContextOfItems)
+    if len(set_instances) > 0:
+        return set_instances
+    else:
+        entity_context = get_entity_contexts(ifc_context, rev_item)
```

Comment:
Changed. I will not close this and the previous conversation, in case you want to take a look at the changed solution. 

I will take a look tomorrow as well, just to be sure. :)